temporal-testing/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -246,9 +252,70 @@ private StartWorkflowExecutionRequest overrideStartWorkflowExecutionRequest(
     if (taskTimeoutMillis != Durations.toMillis(request.getWorkflowTaskTimeout())) {
       request.setWorkflowTaskTimeout(Durations.fromMillis(taskTimeoutMillis));
     }
+
+    SearchAttributes.Builder searchAttributes = request.getSearchAttributes().toBuilder();
```

Comment:
First, it definitely should be in a separate function with a good name. 

But second, I'm not sure what's going on here - do we support only system search attributes here?
We need to provide an ability for our users to define their own attributes and I'm not sure it's what I see here.

Reply:
Refer to this [documentation](https://docs.temporal.io/docs/server/workflow-search/#search-attributes) page: 
"Docker auto-setup image has some pre-allowlisted search attributes that are handy for testing, that have their types indicated in their name:... "

Reply:
@vkoby how did this address the need for users to run tests for workflows with their own custom search attributes?